src/lib.rs

```diff
@@ -269,22 +335,91 @@ impl AnvilChunkMetadata {
     }
 }
 
-impl AnvilRegion {
-    fn new<P: AsRef<Path>>(path: P) -> Result<Self, io::Error> {
-        let mut file = OpenOptions::new()
+pub mod anvil_region {
+    use crate::AnvilChunkMetadata;
+    use bitvec::prelude::*;
+
+    pub fn metadata_index(chunk_x: u8, chunk_z: u8) -> usize {
+        assert!(32 > chunk_x, "Region chunk x coordinate out of bounds");
+        assert!(32 > chunk_z, "Region chunk y coordinate out of bounds");
+
+        chunk_x as usize + chunk_z as usize * 32
+    }
+
+    /// Calculates used sectors.
+    pub fn used_sectors(total_sectors: u32, chunks_metadata: &[AnvilChunkMetadata]) -> BitVec {
+        let mut used_sectors = bitvec![0; total_sectors as usize];
+
+        used_sectors.set(0, true);
+        used_sectors.set(1, true);
+
+        for metadata in chunks_metadata {
+            if metadata.is_empty() {
+                continue;
+            }
+
+            let start_index = metadata.sector_index as usize;
+            let end_index = start_index + metadata.sectors as usize;
+
+            for index in start_index..end_index {
+                used_sectors.set(index, true);
+            }
+        }
+
+        used_sectors
+    }
+}
+
+fn stream_len<S: Seek>(file: &mut S) -> Result<u64, io::Error> {
```

Comment:
Very sad that those methods are not implemented yet. I think it would be better to impl those methods to Seek trait like that:

```rust
trait SeekExt {
    fn stream_len<S: Seek>(file: &mut S) -> Result<u64, io::Error>;
    fn stream_set_len<S: Seek + Write>(file: &mut S, new_len: u64) -> Result<u64, io::Error>;
}

impl<S: Seek> SeekExt for S {
    /// snip
}
```
This will allow call those methods directly from seek.